sui_core/src/authority/authority_store.rs

```diff
@@ -303,14 +307,71 @@ impl<
             .collect())
     }
 
+    // Returns true if the given version of the object has not been deleted (or wrapped).
+    //
+    // Only call this on an object that is already known to exist in self.objects,
+    // (and don't make this method public)
+    fn object_is_alive(&self, object_id: &ObjectID) -> Result<bool, SuiError> {
+        let obj = self
+            .parent_sync
+            .iter()
+            .skip_prior_to(&(*object_id, SequenceNumber::MAX, ObjectDigest::MAX))?
+            .next();
+
+        let obj_ref = match obj {
+            Some((obj_ref, _)) if obj_ref.0 == *object_id => Some(obj_ref),
+            _ => None,
+        };
+
+        match obj_ref {
+            None => {
+                error!(
```

Comment:
I think you want to move the error log out of the function and put it in the None case of calling this function.
Because you also want to cover the case in line 323 where the object should exist but not in the parent_sync.

Reply:
i don't think so - the none case of the calling function is what happens when the object simply doesn't exist, which doesn't indicate an inconsistency.